tensorflow/core/kernels/non_max_suppression_op.cu.cc

```diff
@@ -646,25 +668,103 @@ class NonMaxSuppressionV3GPUOp : public OpKernel {
     }
     const int output_size = max_output_size.scalar<int>()();
     OP_REQUIRES_OK(context, DoNMS(context, boxes, scores, output_size,
-                                  iou_threshold_val, score_threshold_val));
+                                  iou_threshold_val, score_threshold_val,
+                                  /*pad_to_max_output*/ false, &num_boxes));
   }
 };
 
+class NonMaxSuppressionV4GPUOp : public OpKernel {
+ public:
+  explicit NonMaxSuppressionV4GPUOp(OpKernelConstruction* context)
+      : OpKernel(context) {
+    OP_REQUIRES_OK(context, context->GetAttr("pad_to_max_output_size",
+                                             &pad_to_max_output_size_));
+  }
+
+  void Compute(OpKernelContext* context) override {
+    // boxes: [num_boxes, 4]
```

Comment:
Can we add a helper method for these input validation logic as well? They're almost the same for v2/v3/v4 op.